File: phase3/contracts/domain.ts
export type Venue = "solana" | "binance" | "hyperliquid" | "polymarket";
export type Side = "buy" | "sell";
export type OrderType = "market" | "limit";

export interface OrderIntent {
  intentId: string;
  accountId: string;
  strategyId: string;
  venue: Venue;
  instrument: string;
  side: Side;
  orderType: OrderType;
  quantity: string;
  limitPrice?: string;
  maxSlippageBps: number;
  maxFee?: string;
  clientOrderId: string;
  idempotencyKey: string;
  expiresAt: string;
  metadata: Record<string, string>;
}

export interface RiskApproval {
  approvalId: string;
  intentId: string;
  approved: boolean;
  reason: string;
  approvedQuantity: string;
  maxNotional?: string;
  maxSlippageBps: number;
  riskPolicyVersion: string;
  approvedAt: string;
  expiresAt: string;
}

export interface ExecutionRequest {
  executionId: string;
  approvalId: string;
  intentId: string;
  accountId: string;
  venue: Venue;
  instrument: string;
  side: Side;
  quantity: string;
  idempotencyKey: string;
  requestedAt: string;
  authorizationContext: Record<string, string>;
}

export interface FillEvent {
  fillId: string;
  executionId: string;
  venue: Venue;
  venueOrderId?: string;
  venueTradeId?: string;
  instrument: string;
  side: Side;
  quantity: string;
  price: string;
  fee: string;
  feeAsset: string;
  timestamp: string;
  transactionId?: string;
  rawReference?: string;
}

File: phase3/contracts/domain.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum Venue { Solana, Binance, Hyperliquid, Polymarket }

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum Side { Buy, Sell }

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ExecutionRequest {
    pub execution_id: String,
    pub approval_id: String,
    pub intent_id: String,
    pub account_id: String,
    pub venue: Venue,
    pub instrument: String,
    pub side: Side,
    pub quantity: String,
    pub idempotency_key: String,
    pub requested_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FillEvent {
    pub fill_id: String,
    pub execution_id: String,
    pub venue: Venue,
    pub venue_order_id: Option<String>,
    pub venue_trade_id: Option<String>,
    pub instrument: String,
    pub side: Side,
    pub quantity: String,
    pub price: String,
    pub fee: String,
    pub fee_asset: String,
    pub timestamp: String,
    pub transaction_id: Option<String>,
}
